src/libraries/Reserves.sol

```diff
@@ -7,36 +7,46 @@ import {CustomRevert} from "./CustomRevert.sol";
 library Reserves {
     using CustomRevert for bytes4;
 
+    /// @notice Thrown when a user has already synced a currency, but not yet settled
+    error AlreadySynced();
+
     /// bytes32(uint256(keccak256("ReservesOf")) - 1)
     bytes32 constant RESERVES_OF_SLOT = 0x1e0745a7db1623981f0b2a5d4232364c00787266eb75ad546f190e6cebe9bd95;
-    /// @notice The transient reserves for pools with no balance is set to the max as a sentinel to track that it has been synced.
-    uint256 public constant ZERO_BALANCE = type(uint256).max;
+    /// bytes32(uint256(keccak256("Sync")) - 1)
+    bytes32 constant SYNC_SLOT = 0xf0e14a408baf7f453312eec68e9b7d728ec5337fbdf671f917ee8c80f3255231;
 
-    /// @notice Thrown when someone has not called sync before calling settle for the first time.
-    error ReservesMustBeSynced();
+    function checkNotSynced() internal view {
+        Currency syncing;
+        assembly {
+            syncing := tload(SYNC_SLOT)
+        }
+        if (!syncing.isZero()) {
+            AlreadySynced.selector.revertWith();
+        }
+    }
 
-    function setReserves(Currency currency, uint256 value) internal {
-        if (value == 0) value = ZERO_BALANCE;
-        bytes32 key = _getKey(currency);
+    function getSyncedCurrency() internal view returns (Currency currency) {
         assembly {
-            tstore(key, value)
+            currency := tload(SYNC_SLOT)
         }
     }
 
-    function getReserves(Currency currency) internal view returns (uint256 value) {
-        bytes32 key = _getKey(currency);
+    function reset() internal {
```

Comment:
can you add a comment here that the reserves of slot will be dirtied, but it is safe because it is always overridden before the next settle... or just to be extra safe we should wipe it even in reset (in which case you can use Reserves.setReserves(0, 0);